packages/jest-mock-apollo/src/types.ts

```diff
@@ -0,0 +1,19 @@
+import {GraphQLRequest} from 'apollo-link';
+import {GraphQLSchema} from 'graphql';
+import {ApolloReducerConfig} from 'apollo-cache-inmemory';
+
+export type MockGraphQLResponse = Error | object;
+export type GraphQLMock =
+  | {[key: string]: MockGraphQLResponse}
+  | ((request: GraphQLRequest) => MockGraphQLResponse);
+
+export interface GraphQLClientConfig {
+  unionOrIntersectionTypes?: any[];
+  schema?: GraphQLSchema;
+  schemaSrc?: string;
```

Comment:
IMO it is not good API design, especially initially, to expose two ways of accomplishing identical things. We should pick the thing we know addresses the use cases we consider in scope (in this case, some projects have just the JSON from introspection, some have the schema .graphql file, so the `schema: GraphQLSchema` option is the one they can both use).

Reply:
fair enough. I was trying to make the user of our package do less work, but I suppose calling `buildSchema` is pretty trivial.